### PR TITLE
fix: Add UniswapX and Relay event watchers

### PR DESCRIPTION
**Pull request by zhongeric**
Needed for order service integration. Creates two separate classes for event watching, one for UniswapX-style orders and one for Relay orders. This is needed because they use different contracts which have different fill event schemas. The export for EventWatcher has been removed and instead you must import either UniswapX/RelayEventWatcher explicity.

Note that the behavior is slightly different between the UniswapX event watcher and the Relay event watcher:

In UniswapX,
- inputs are tokens that are sent to the filler (input tokens from the order)
- outputs are tokens that are sent to the swapper (output tokens from the order)

In relayer,
- inputs are tokens sent to the filler (the feeToken paid by the swapper)
- outputs are tokens sent to the swapper (the output amount from the swap)

This PR also fixes broken unit and integ tests on main

### CODE REVIEW

File: src/utils/EventWatcher.ts [hunk 1]
```diff
@@ -120,22 +149,39 @@ export class EventWatcher {
...
+export class RelayEventWatcher extends EventWatcher<RelayOrderReactor> {
+  constructor(provider: BaseProvider, reactorAddress: string) {
+    const reactor = RelayOrderReactor__factory.connect(
+      reactorAddress,
+      provider
+    );
+    super(reactor);
+  }
+
+  async getFillLogs(fromBlock: number, toBlock?: number): Promise<FillEvent[]> {
```
Comment (robert-seitz-uniswap) [inline src/utils/EventWatcher.ts:163]: looks like we are using getFillInfo in the service. Is there a difference in implementation form that and getFillLogs?

Comment (robert-seitz-uniswap) [reply src/utils/EventWatcher.ts:163]: reference
```
export declare type TokenTransfer = {
    token: string;
    amount: BigNumber;
};
export interface FillData {
    orderHash: string;
    filler: string;
    nonce: BigNumber;
    swapper: string;
}
export interface FillInfo extends FillData {
    blockNumber: number;
    txHash: string;
    inputs: TokenTransfer[];
    outputs: TokenTransfer[];
}
/**
 * Helper for watching events
 */
export declare class EventWatcher {
    private reactor;
    constructor(provider: BaseProvider, reactorAddress: string);
    getFillLogs(fromBlock: number, toBlock?: number): Promise<FillEvent[]>;
    getFillEvents(fromBlock: number, toBlock?: number): Promise<FillData[]>;
    getFillInfo(fromBlock: number, toBlock?: number): Promise<FillInfo[]>;
    onFill(callback: (fillData: FillData, event: Event) => void): void;
    getTokenTransfers(logs: Log[], recipient: string): {
        token: string;
        amount: BigNumber;
    }[];
}

```

File: src/utils/EventWatcher.ts [hunk 2]
```diff
@@ -104,6 +97,42 @@ export class EventWatcher {
...
+        }
+        return logAcc;
+      } catch (e) {
+        return logAcc;
+      }
+    }, [] as { token: string; amount: BigNumber }[]);
+  }
+}
+
+export class UniswapXEventWatcher extends EventWatcher<ExclusiveDutchOrderReactor> {
```
Comment (ConjunctiveNormalForm) [inline src/utils/EventWatcher.ts:119]: we should be able to extend this to support X v2?

**Review by ConjunctiveNormalForm — APPROVED** [review]
lgtm! I think the UniswapXEventWatcher can support XV2 out of the box if we generalize the Reactor type?

File: src/utils/EventWatcher.ts (hunk 1 above)
Comment (zhongeric) [reply src/utils/EventWatcher.ts:163]: getFillLogs is a function called by getFillInfo, it gets the raw logs on chain and fill info adds order specific data

File: src/utils/EventWatcher.ts (hunk 2 above)
Comment (zhongeric) [reply src/utils/EventWatcher.ts:119]: Yes it should be able to, can try in a follow up PR

**Pull request merged**